packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/ScatterPlotViewer/components/ScatterPlot/components/Selections/Selections.js

```diff
@@ -0,0 +1,168 @@
+  height,
+  margin,
+  width,
+  xScale,
+  yScale
+}) {
+  const { onKeyZoom } = useKeyZoom()
+  const eventRoot = useRef()
+  const [selections, setSelections] = useState([])
+  const [activeIndex, setActiveIndex] = useState(-1)
+  const [createNewBrush, setCreateNewBrush] = useState(true)
+  const activeSelection = selections[activeIndex]
+  
+  let key = 'new-brush'
+  let initialBrushPosition
+  if (activeSelection) {
+    const start = xScale(activeSelection.x0)
+    const end = xScale(activeSelection.x1)
+    initialBrushPosition = {
+      start: { x: start },
+      end: { x: end }
+    }
+    key = `active-selection-${start}-${end}`
+  }
+
+  if (disabled && activeSelection) {
+    setActiveIndex(-1)
+    setCreateNewBrush(true)
+  }
+
+  function deleteSelection(index) {
+    const newSelections = selections.filter((selection, i) => i !== index)
+    setSelections(newSelections)
+    if (index === activeIndex) {
+      setActiveIndex(-1)
+      setCreateNewBrush(true)
+    }
+  }
+
+  function onKeyDown(event) {
+    switch (event.key) {
+      case 'Backspace': {
+        if (activeIndex > -1) {
+          event.preventDefault()
+          deleteSelection(activeIndex)
+          return false
+        }
+      }
+    }
+    return onKeyZoom(event)
+  }
+
+  function onStartNewBrush() {
+    setCreateNewBrush(true)
+  }
+
+  function onSelectBrush(index) {
+    setActiveIndex(index)
+    setCreateNewBrush(false)
+  }
+
+  function onStartSelection({ x }) {
+    let index = -1
+    selections.forEach((selection, i) => {
+      const lower = xScale(selection.x0) - HANDLE_SIZE
+      const upper = xScale(selection.x1) + HANDLE_SIZE
+      const click = xScale(x)
+      const isSelection = (click > lower) && (click < upper)
+      if (isSelection) {
+        index = i
+      }
+    })
+    if (index > -1) {
+      onSelectBrush(index)
+    } else {
+      onStartNewBrush()
+    }
+  }
+
+  function createNewSelection(selection) {
+    if (selection) {
+      const newSelections = [...selections, selection]
+      setSelections(newSelections)
+      return newSelections.length - 1
+    } else {
+      return -1
+    }
+  }
+
+  function replaceActiveSelection(selection) {
+    if (selection) {
+      const newSelections = [...selections]
+      newSelections[activeIndex] = selection
+      setSelections(newSelections)
+      return activeIndex
+    } else {
+      setCreateNewBrush(true)
+      return -1
+    }
+  }
+
+  function onEndSelection(selection) {
+    const newSelection = createNewBrush ? createNewSelection(selection) : replaceActiveSelection(selection)
+    setActiveIndex(newSelection)
+    eventRoot?.current.focus()
+  }
+
+  return (
+    <g
+      ref={eventRoot}
+      focusable='true'
+      onKeyDown={onKeyDown}
+      tabIndex='-1'
+    >
+      {selections.map((selection, index) => (
+        <Selection
+          key={`selection-${index}`}
+          disabled={!createNewBrush && selection === activeSelection}
```

Comment:
This is confusing, because it disables the active selection. I think selections should be disabled when the parent component is disabled, and active state should be tracked separately from disabled state.